### PR TITLE
chore: update common module

### PR DESCRIPTION
**Pull request by KraMorK**
## Description

Reduce differences between cse and main branches. Standardize build, update module dependencies
## Why do we need it, and what problem does it solve?

Migrate cse changes to main branch
## Why do we need it in the patch release (if we do)?

not related to release
## What is the expected result?

Migrate cse changes to main branch
## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

```changes
section: common
type: chore
summary: Reduce differences between cse and main branches - common-images.
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/eks'...

**Comment by KraMorK** [comment]
/e2e/abort eks;WithoutNAT;containerd;1.27 11957541171 11957541171-1-con-1-27 /sys/deckhouse-oss/install:pr10753

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/eks'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/aws'...

**Comment by Taior** [comment]
/e2e/abort eks;WithoutNAT;containerd;1.27 12046506974 12046506974-1-con-1-27 /sys/deckhouse-oss/install:pr10753

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Taior. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/eks'...

File: modules/000-common/images/check-kernel-version/werf.inc.yaml [hunk 1]
```diff
@@ -9,14 +9,22 @@ import:
...
+artifact: {{ $.ModuleName }}/{{ $.ImageName }}-src-artifact
+fromArtifact: common/src-artifact
 git:
 - add: /{{ $.ModulePath }}modules/000-{{ $.ModuleName }}/images/{{ $.ImageName }}/src
   to: /src
   stageDependencies:
     install:
     - '**/*'
+---
+artifact: {{ .ModuleName }}/{{ .ImageName }}-artifact
```
Comment (name212) [inline modules/000-common/images/check-kernel-version/werf.inc.yaml:21]:
```suggestion
artifact: {{ $.ModuleName }}/{{ $.ImageName }}-artifact
```

File: modules/000-common/images/check-kernel-version/werf.inc.yaml [hunk 2]
```diff
@@ -9,14 +9,22 @@ import:
...
+from: {{ .Images.BASE_GOLANG_23_ALPINE }}
```
Comment (name212) [inline modules/000-common/images/check-kernel-version/werf.inc.yaml:22]:
```suggestion
from: {{ $.Images.BASE_GOLANG_23_ALPINE }}
```

File: modules/000-common/images/csi-external-provisioner/werf.inc.yaml [hunk 3]
```diff
@@ -6,35 +6,46 @@
...
+  to: /patches
+  stageDependencies:
+    install:
+    - '**/*'
+shell:
+  install:
+  - git clone --depth 1 --branch {{ $value.csi.provisioner }} {{ $.SOURCE_REPO }}/kubernetes-csi/external-provisioner.git /src
+  - cd /src
+  - test -d "/patches/{{ $value.csi.provisioner }}" && git apply /patches/{{ $value.csi.provisioner }}/*.patch --verbose
+  - rm -r vendor
```
Comment (name212) [inline modules/000-common/images/csi-external-provisioner/werf.inc.yaml:29]: Why do we need remove vendor here?

File: modules/000-common/images/iptables-wrapper/werf.inc.yaml [hunk 4]
```diff
@@ -7,16 +7,26 @@ import:
   to: /iptables-wrapper
   before: setup
 ---
+artifact: {{ $.ModuleName }}/{{ $.ImageName }}-src-artifact
+fromArtifact: common/src-artifact
+shell:
+  install:
+  - git clone {{ .SOURCE_REPO }}/kubernetes-sigs/iptables-wrappers.git /src
```
Comment (name212) [inline modules/000-common/images/iptables-wrapper/werf.inc.yaml:14]:
```suggestion
  - git clone {{ $.SOURCE_REPO }}/kubernetes-sigs/iptables-wrappers.git /src
```

File: modules/000-common/images/kube-rbac-proxy/werf.inc.yaml [hunk 5]
```diff
@@ -10,29 +10,38 @@ docker:
...
     install:
     - '**/*'
+shell:
+  install:
+  - git clone --depth 1 --branch v0.11.0 {{ .SOURCE_REPO }}/brancz/kube-rbac-proxy.git /src
+  - cd /src
+  - git apply /patches/*.patch --verbose
+  - rm -r .git
+---
+artifact: {{ .ModuleName }}/{{ .ImageName }}-artifact
```
Comment (name212) [inline modules/000-common/images/kube-rbac-proxy/werf.inc.yaml:28]:
```suggestion
artifact: {{ $.ModuleName }}/{{ $.ImageName }}-artifact
```

File: modules/000-common/images/nginx-static/werf.inc.yaml [hunk 6]
```diff
@@ -1,14 +1,30 @@
...
+fromArtifact: common/src-artifact
+shell:
+  install:
+  - git clone -b release-{{ $NginxVersion }} --single-branch --depth=1 {{ $.SOURCE_REPO }}/nginx/nginx.git /src
+  - cd /src
+  - rm -r .git
+  - git clone -b openssl-{{ $OpenSSLVersion }} --single-branch --depth=1 {{ $.SOURCE_REPO }}/openssl/openssl.git
+  - rm -r openssl/.git
 ---
 artifact: {{ .ModuleName }}/build-nginx-static-artifact
```
Comment (name212) [inline modules/000-common/images/nginx-static/werf.inc.yaml:14]:
```suggestion
artifact: {{ $.ModuleName }}/build-nginx-static-artifact
```

File: modules/000-common/images/nginx-static/werf.inc.yaml [hunk 7]
```diff
@@ -1,14 +1,30 @@
...
-from: {{ .Images.BASE_UBUNTU_DEV }}
+from: {{ .Images.BASE_UBUNTU }}
```
Comment (name212) [inline modules/000-common/images/nginx-static/werf.inc.yaml:15]:
```suggestion
from: {{ $.Images.BASE_UBUNTU }}
```

File: modules/000-common/images/shell-operator/werf.inc.yaml [hunk 8]
```diff
@@ -33,31 +32,54 @@ import:
...
         url = {{ .SOURCE_REPO }}/flant/oniguruma.git
     EOF
   - git submodule update --init
+  - rm -r docs
+  - rm -r /src/jq/.git
+  - rm -r /src/jq/modules/oniguruma/.git
+  - mkdir -p docs/content/manual
+  - touch docs/content/manual/manual.yml
+---
+artifact: {{ .ModuleName }}/{{ $.ImageName }}-artifact
```
Comment (name212) [inline modules/000-common/images/shell-operator/werf.inc.yaml:56]:
```suggestion
artifact: {{ $.ModuleName }}/{{ $.ImageName }}-artifact
```

File: modules/000-common/images/shell-operator/werf.inc.yaml [hunk 9]
```diff
@@ -1,5 +1,4 @@
 {{- $shellOperatorVersion := "v1.5.1"}}
-{{- $jqVersion := "b6be13d5"}}
 ---
 image: {{ .ModuleName }}/{{ $.ImageName }}
```
Comment (name212) [inline modules/000-common/images/shell-operator/werf.inc.yaml:3]:
```suggestion
image: {{ $.ModuleName }}/{{ $.ImageName }}
```

File: modules/000-common/images/shell-operator/werf.inc.yaml [hunk 10]
```diff
@@ -33,31 +32,54 @@ import:
...
 shell:
-  beforeInstall:
-  - apk update
-  - apk add --no-cache automake autoconf libtool build-base git
   install:
-  - export GOPROXY={{ .GOPROXY }} CGO_ENABLED=1 GOOS=linux
-  - git clone {{ .SOURCE_REPO }}/flant/jq.git
-  - cd /jq
-  - git reset --hard {{ $jqVersion }}
+  - git clone --branch {{ $shellOperatorVersion }} --depth 1 {{ .SOURCE_REPO }}/flant/shell-operator.git /src/shell-operator
```
Comment (name212) [inline modules/000-common/images/shell-operator/werf.inc.yaml:39]:
```suggestion
  - git clone --branch {{ $shellOperatorVersion }} --depth 1 {{ $.SOURCE_REPO }}/flant/shell-operator.git /src/shell-operator
```

File: modules/000-common/images/shell-operator/werf.inc.yaml [hunk 11]
```diff
@@ -33,31 +32,54 @@ import:
...
+  - rm -r /src/shell-operator/.git
+  - git clone --depth 1 {{ .SOURCE_REPO }}/jqlang/jq --branch "jq-1.7.1" /src/jq
```
Comment (name212) [inline modules/000-common/images/shell-operator/werf.inc.yaml:41]:
```suggestion
  - git clone --depth 1 {{ $.SOURCE_REPO }}/jqlang/jq --branch "jq-1.7.1" /src/jq
```

**Review by name212 — CHANGES_REQUESTED** [review]
Some nits

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/aws'...

**Comment by KraMorK** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 12155676965 12155676965-1-con-1-27 /sys/deckhouse-oss/install:pr10753 ec2-user@52.58.158.9

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @KraMorK. I've started the workflow for label 'e2e/run/aws'...

**Review by name212 — APPROVED** [review]

**Pull request merged**